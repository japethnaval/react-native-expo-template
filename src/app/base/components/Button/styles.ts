import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  base: {
    borderRadius: 5,
    padding: 16,
  },
  text: {
    color: '#fff',
    textAlign: 'center',
  },
  danger: {
    backgroundColor: 'red',
  },
  success: {
    backgroundColor: 'green',
  },
  warning: {
    backgroundColor: 'orange',
  },
  default: {
    backgroundColor: 'blue',
  },
})

export default styles
