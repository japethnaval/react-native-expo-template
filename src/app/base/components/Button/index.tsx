import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'
import styles from './styles'

type IVariant = 'danger' | 'success' | 'warning' | 'default'

interface IProps {
  text: string
  variant?: IVariant
}

export const Button: React.FC<IProps> = ({ text, variant = 'default' }) => {
  return (
    <TouchableOpacity style={StyleSheet.flatten([styles.base, styles[variant]])}>
      <Text style={styles.text}> {text} </Text>
    </TouchableOpacity>
  )
}

export default Button
