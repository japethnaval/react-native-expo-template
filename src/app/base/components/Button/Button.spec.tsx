import React from 'react'
import { shallow } from 'enzyme'
import Button from './index'

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Button text="test label" />)
      expect(component).toMatchSnapshot()
    })
  })
})
