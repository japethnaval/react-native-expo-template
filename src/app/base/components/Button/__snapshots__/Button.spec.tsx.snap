// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<ForwardRef
  style={
    Object {
      "backgroundColor": "blue",
      "borderRadius": 5,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "textAlign": "center",
      }
    }
  >
     
    test label
     
  </Text>
</ForwardRef>
`;
